src/Pos/Txp/Core/Tx.hs

```diff
@@ -244,6 +245,10 @@ verifyTxDo verifyAlone verifyVersions _gContext extendedInputs
         | otherwise =
               let txSigData = (txInHash, txInIndex, txOutHash, distrsHash)
               in txScriptCheck txSigData twValidator twRedeemer
+    validateTxIn _i TxIn{..} _ RedeemWitness{..} =
+        if redeemCheckSig twRedeemKey (txInHash, txInIndex, txOutHash, distrsHash) twRedeemSig
```

Comment:
Why not use guards? Not only it's generally preferred in code, but also it's used in other patterns for this function.